lib/utils/mailup.ts

```diff
@@ -0,0 +1,287 @@
+}
+
+async function sendTransactionalMail(
+  creds: SmtpAuthInfo,
+  payload: EmailPayload
+): Promise<Either<Error, ApiResponse>> {
+  const errorOrResponse = await callMailUpApi(
+    "POST",
+    ENDPOINTS.sendTransactionalMail,
+    creds,
+    payload
+  );
+  if (isLeft(errorOrResponse)) {
+    return errorOrResponse;
+  } else {
+    const response = errorOrResponse.value;
+    if (response && response.Code && response.Code === "0") {
+      return right(response);
+    } else {
+      return left(
+        new Error(
+          `Error while sending email: ${response.Code} - ${response.Message}`
+        )
+      );
+    }
+  }
+}
+
+function toMailupAddresses(
+  addresses: ReadonlyArray<NodemailerAddress> | undefined
+): Option<ReadonlyArray<Address>> {
+  return Array.isArray(addresses) && addresses.length > 0
+    ? some(
+        addresses.map((address: NodemailerAddress) => {
+          return {
+            Email: t.validate(address.address, EmailString).fold(() => {
+              // this never happens as nodemailer has already parsed the address
+              throw new Error(
+                `Error while parsing email address (toMailupAddresses): invalid format '${
+                  address.address
+                }'.`
+              );
+            }, t.identity),
+            Name: address.name || address.address
+          };
+        })
+      )
+    : none;
+}
+
+function toMailupAddress(
+  addresses: ReadonlyArray<NodemailerAddress> | undefined
+): Option<Address> {
+  const addrs = toMailupAddresses(addresses).toUndefined();
+  return addrs ? fromNullable(addrs[0]) : none;
+}
+
+/**
+ * Nodemailer transport for MailUp transactional APIs
+ *
+ * see http://help.mailup.com/display/mailupapi/Transactional+Emails+using+APIs
+ * and https://nodemailer.com/plugins/create/#transports
+ *
+ * Usage:
+ *
+ * const transporter = nodemailer.createTransport(
+ *   MailUpTransport({
+ *     creds: {
+ *       Username: <SMPT+Username>,
+ *       Secret: <SMPT+Password>
+ *     }
+ *   })
+ * );
+ *
+ * transporter
+ *   .sendMail({
+ *     from:      "foobar@xexample.com",
+ *     to:        "deadbeef@xexample.com",
+ *     replyTo:   "foobar-reply@xexample.com",
+ *     subject:   "lorem ipsum",
+ *     text:      "lorem ipsum",
+ *     html:      "<b>lorem ipsum</b>"
+ *   })
+ *   .then(res => console.log(JSON.stringify(res)))
+ *   .catch(err => console.error(JSON.stringify(err)));
+ */
+export function MailUpTransport(
+  options: IMailUpTransportOptions
+): nodemailer.Transport {
+  return {
+    name: TRANSPORT_NAME,
+
+    version: TRANSPORT_VERSION,
+
+    send: async (mail, callback) => {
+      // We don't use mail.data.from / mail.data.to as they are not parsed.
+      // The following cast exists because of a bug in nodemailer typings.
+      const addresses: IAddresses = mail.message.getAddresses() as IAddresses;
+
+      // To convert nodemailer streams to strings:
```

Comment:
if this code is not needed, remove it

Reply:
is not clear from the nodemailer documentation if this is not needed. by now I've observed that 
mail.data.text and mail.data.subject are just plain text, but if you look at the nodemailer docs and typings it seems (but I couldn't verify it) that they CAN (in some **unspecified scenarios**) be streams that must be converted to strings before passing them to the MailUP API.